src/lib/Components/NavigationHub.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace EzSystems\HybridPlatformUi\Components;
+
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class NavigationHub implements Component
+{
+    const TAG_NAME = 'ez-navigation-hub';
+
+    const ACTIVE_ZONE_CLASS = 'ez-active-zone';
+
+    const MATCHED_LINK_CLASS = 'ez-matched-link';
+
+    /**
+     * @var \Twig_Environment
+     */
+    protected $templating;
+
+    /**
+     * @var \EzSystems\HybridPlatformUi\NavigationHub\Zone[]
+     */
+    protected $zones;
+
+    /**
+     * @var \EzSystems\HybridPlatformUi\NavigationHub\Link[]
+     */
+    protected $links;
+
+    /**
+     * @var \Symfony\Component\HttpFoundation\Request
+     */
+    protected $request;
+
+    public function __construct($templating, RequestStack $stack, array $zones = [], array $links = [])
+    {
+        $this->request = $stack->getMasterRequest();
+        $this->templating = $templating;
+        $this->zones = $zones;
+        $this->links = $links;
+    }
+
+    public function __toString()
+    {
+        return $this->templating->render(
+            'EzSystemsHybridPlatformUiBundle:components:navigationhub.html.twig',
+            [
+                'tag' => self::TAG_NAME,
```

Comment:
But in any case `static` should always be preferred to `self` as it's way more flexible, without significant impact on performance